src/style/theme/variants/default/spacing.ts

```diff
@@ -1,126 +1,128 @@
-import SPACING from '../../spacingPallette';
+import ThemeContextType from '../../../../types/provider/themeContextType';
 
-const spacing = {
-  m1: {
-    margin: SPACING[1]
-  },
-  m2: {
-    margin: SPACING[2]
-  },
-  m3: {
-    margin: SPACING[3]
-  },
-  m4: {
-    margin: SPACING[4]
-  },
-  m5: {
-    margin: SPACING[5]
-  },
-  m6: {
-    margin: SPACING[6]
-  },
-  m7: {
-    margin: SPACING[7]
-  },
-  m8: {
-    margin: SPACING[8]
-  },
-  mr1: {
-    marginRight: SPACING[1]
-  },
-  mr2: {
-    marginRight: SPACING[2]
-  },
-  mr3: {
-    marginRight: SPACING[3]
-  },
-  mr4: {
-    marginRight: SPACING[4]
-  },
-  mr5: {
-    marginRight: SPACING[5]
-  },
-  mr6: {
-    marginRight: SPACING[6]
-  },
-  mr7: {
-    marginRight: SPACING[7]
-  },
-  mr8: {
-    marginRight: SPACING[8]
-  },
-  ml1: {
-    marginLeft: SPACING[1]
-  },
-  ml2: {
-    marginLeft: SPACING[2]
-  },
-  ml3: {
-    marginLeft: SPACING[3]
-  },
-  ml4: {
-    marginLeft: SPACING[4]
-  },
-  ml5: {
-    marginLeft: SPACING[5]
-  },
-  ml6: {
-    marginLeft: SPACING[6]
-  },
-  ml7: {
-    marginLeft: SPACING[7]
-  },
-  ml8: {
-    marginLeft: SPACING[8]
-  },
-  mt1: {
-    marginTop: SPACING[1]
-  },
-  mt2: {
-    marginTop: SPACING[2]
-  },
-  mt3: {
-    marginTop: SPACING[3]
-  },
-  mt4: {
-    marginTop: SPACING[4]
-  },
-  mt5: {
-    marginTop: SPACING[5]
-  },
-  mt6: {
-    marginTop: SPACING[6]
-  },
-  mt7: {
-    marginTop: SPACING[7]
-  },
-  mt8: {
-    marginTop: SPACING[8]
-  },
-  mb1: {
-    marginBottom: SPACING[1]
-  },
-  mb2: {
-    marginBottom: SPACING[2]
-  },
-  mb3: {
-    marginBottom: SPACING[3]
-  },
-  mb4: {
-    marginBottom: SPACING[4]
-  },
-  mb5: {
-    marginBottom: SPACING[5]
-  },
-  mb6: {
-    marginBottom: SPACING[6]
-  },
-  mb7: {
-    marginBottom: SPACING[7]
-  },
-  mb8: {
-    marginBottom: SPACING[8]
-  }
+const spacing = (SPACING_PALETTE: ThemeContextType['spacing']) => {
+  return {
+    m1: {
+      margin: SPACING_PALETTE[1]
```

Comment:
maybe you forgot the `!` 
`margin: SPACING_PALETTE[1]`
==>
`margin: SPACING_PALETTE![1]`

Reply:
In our case, the `not null` operator is just for sub objects to prevent ts errors ;)